### PR TITLE
[WIP] - December 2022 rke2 - dev-v2.6-1222-patches

### PR DESCRIPTION
**Pull request by briandowns**

### CODE REVIEW

File: channels-rke2.yaml [hunk 1]
```diff
@@ -639,6 +639,14 @@ releases:
     charts: &charts-v1-22-16-rke2r1
       <<: *charts-v1-22-15-rke2r2
     featureVersions: *featureVersions-v1
+  - version: v1.22.17-rc1+rke2r1
+    minChannelServerVersion: v2.6.3-alpha1
+    maxChannelServerVersion: v2.6.99
+    serverArgs: *serverArgs-v1-22-11-rke2r1
+    agentArgs: *agentArgs-v1-22-11-rke2r1
```
Comment (matttrach) [inline channels-rke2.yaml:646]: I see several releases are pointing to &agentArgs-v1-22-11-rke2r1 , did something significant happen in that release? Why not use &agentArgs-v1-22-15-rke2r2? Should we generate a new anchor with every release?

The inheritance goes like this:
&agentArgs-v1-22-15-rke2r2 -> 
&agentArgs-v1-22-13-rke2r1 -> 
&agentArgs-v1-22-11-rke2r1 ->
&agentArgs-v1-22-10-rke2r2 ->
&agentArgs-v1-22-9-rke2r2 ->
&agentArgs-v1-22-9-rke2r1 ->
&agentArgs-v1-22-7-rke2r2 ->
&agentArgs-v1-22-6-rke2r1 ->
&agentArgs-v1-22-5-rke2r2 ->
&agentArgs-v1-22-4-rke2r2 ->
&agentArgs-v1-21-5-rke2r2 ->
&agentArgs-v1-21-5-rke2r1 (which has merged changes)

File: channels-rke2.yaml [hunk 2]
```diff
@@ -639,6 +639,14 @@ releases:
     charts: &charts-v1-22-16-rke2r1
       <<: *charts-v1-22-15-rke2r2
     featureVersions: *featureVersions-v1
+  - version: v1.22.17-rc1+rke2r1
+    minChannelServerVersion: v2.6.3-alpha1
+    maxChannelServerVersion: v2.6.99
+    serverArgs: *serverArgs-v1-22-11-rke2r1
```
Comment (matttrach) [inline channels-rke2.yaml:645]: I see several releases are pointing to &serverArgs-v1-22-11-rke2r1, did something significant happen in that release? Why not use &serverArgs-v1-22-15-rke2r2? Should we generate a new anchor with every release?

The inheritance goes like this:
&serverArgs-v1-22-15-rke2r2 ->
&serverArgs-v1-22-13-rke2r1 ->
&serverArgs-v1-22-11-rke2r1 ->
&serverArgs-v1-22-10-rke2r2 (which has merged changes)

File: channels-rke2.yaml [hunk 3]
```diff
@@ -833,6 +841,34 @@ releases:
       rke2-calico-crd:
         repo: rancher-rke2-charts
         version: v3.24.103
+  - version: v1.23.15-rc1+rke2r1
+    minChannelServerVersion: v2.6.4-alpha1
+    maxChannelServerVersion: v2.6.99
+    serverArgs: *serverArgs-v1-23-8-rke2r1
+    agentArgs: *agentArgs-v1-23-8-rke2r1
```
Comment (matttrach) [inline channels-rke2.yaml:851]: I see several releases are pointing to &agentArgs-v1-23-8-rke2r1, did something significant happen in that release? Why not use &agentArgs-v1-23-10-rke2r1? Should we generate a new anchor with every release?

The inheritance model for this goes:
&agentArgs-v1-23-10-rke2r1 ->
&agentArgs-v1-23-8-rke2r1 ->
&agentArgs-v1-23-7-rke2r2 ->
&agentArgs-v1-23-6-rke2r2 ->
&agentArgs-v1-23-6-rke2r1 ->
&agentArgs-v1-23-4-rke2r2 ->
&agentArgs-v1-22-7-rke2r2 ->
&agentArgs-v1-22-6-rke2r1 ->
&agentArgs-v1-22-5-rke2r2 ->
&agentArgs-v1-22-4-rke2r2 ->
&agentArgs-v1-21-5-rke2r2 ->
&agentArgs-v1-21-5-rke2r1 (which has merged changes)

File: channels-rke2.yaml [hunk 4]
```diff
@@ -833,6 +841,34 @@ releases:
       rke2-calico-crd:
         repo: rancher-rke2-charts
         version: v3.24.103
+  - version: v1.23.15-rc1+rke2r1
+    minChannelServerVersion: v2.6.4-alpha1
+    maxChannelServerVersion: v2.6.99
+    serverArgs: *serverArgs-v1-23-8-rke2r1
```
Comment (matttrach) [inline channels-rke2.yaml:850]: I see several releases are pointing to &serverArgs-v1-23-8-rke2r1, did something significant happen in that release? Why not use &serverArgs-v1-23-10-rke2r1? Should we generate a new anchor with every release?

The inheritance model for this goes:
&serverArgs-v1-23-10-rke2r1 ->
&serverArgs-v1-23-8-rke2r1 ->
&serverArgs-v1-23-7-rke2r2 ->
&serverArgs-v1-23-6-rke2r2 ->
&serverArgs-v1-23-6-rke2r1 ->
&serverArgs-v1-23-4-rke2r2 ->
&serverArgs-v1-22-7-rke2r2 ->
&serverArgs-v1-22-6-rke2r1 ->
&serverArgs-v1-22-5-rke2r2 ->
&serverArgs-v1-22-4-rke2r2 ->
&serverArgs-v1-21-7-rke2r2 (which has merged changes)

File: channels-rke2.yaml [hunk 5]
```diff
@@ -928,3 +964,32 @@ releases:
         repo: rancher-rke2-charts
         version: v3.24.103
     featureVersions: *featureVersions-v1
+  - version: v1.24.9-rc1+rke2r1
+    minChannelServerVersion: v2.6.7-alpha1
+    maxChannelServerVersion: v2.6.99
+    serverArgs: *serverArgs-v1-24-2-rke2r1
```
Comment (matttrach) [inline channels-rke2.yaml:973]: I see several releases are pointing to &serverArgs-v1-24-2-rke2r1, did something significant happen in that release? Why not use &serverArgs-v1-24-4-rke2r1? Should we generate a new anchor with every release?

The inheritance model goes like this:
&serverArgs-v1-24-4-rke2r1 ->
&serverArgs-v1-24-2-rke2r1 ->
&serverArgs-v1-23-8-rke2r1 ->
&serverArgs-v1-23-7-rke2r2 ->
&serverArgs-v1-23-6-rke2r2 ->
&serverArgs-v1-23-6-rke2r1 ->
&serverArgs-v1-23-4-rke2r2 ->
&serverArgs-v1-22-7-rke2r2 ->
&serverArgs-v1-22-6-rke2r1 ->
&serverArgs-v1-22-5-rke2r2 ->
&serverArgs-v1-22-4-rke2r2 ->
&serverArgs-v1-21-7-rke2r2 (which has merged changes)

File: channels-rke2.yaml [hunk 6]
```diff
@@ -928,3 +964,32 @@ releases:
...
+    agentArgs: *agentArgs-v1-24-2-rke2r1
```
Comment (matttrach) [inline channels-rke2.yaml:974]: I see several releases are pointing to &agentArgs-v1-24-2-rke2r1 , did something significant happen in that release? Why not use &agentArgs-v1-24-4-rke2r1 ? Should we generate a new anchor with every release?

The inheritance model goes like this:
&agentArgs-v1-24-4-rke2r1 ->
&agentArgs-v1-24-2-rke2r1 ->
&agentArgs-v1-23-8-rke2r1 ->
&agentArgs-v1-23-7-rke2r2 ->
&agentArgs-v1-23-6-rke2r2 ->
&agentArgs-v1-23-6-rke2r1 ->
&agentArgs-v1-23-4-rke2r2 ->
&agentArgs-v1-22-7-rke2r2 ->
&agentArgs-v1-22-6-rke2r1 ->
&agentArgs-v1-22-5-rke2r2 ->
&agentArgs-v1-22-4-rke2r2 ->
&agentArgs-v1-21-5-rke2r2 ->
&agentArgs-v1-21-5-rke2r1 (which has merged changes)

**Review by matttrach — DISMISSED** [review]
I have questions, but everything is valid.

**Comment by rancher-max** [comment]
https://github.com/rancher/rancher/issues/39887

**Review by rancher-max — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Pull request merged**